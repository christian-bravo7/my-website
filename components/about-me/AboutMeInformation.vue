<template>
  <section class="section home__content--secondary min-screen who-iam-section">
    <div class="who-iam-section__container container">
      <div class="who-iam-section__content is-medium">
        <h1 class="who-iam-section__title">
          {{ $t('me.i-label') }}
        </h1>
        <p class="who-iam-section__description">
          {{ $t('me.about-me-description') }}
        </p>
      </div>
      <div class="who-iam-section__asset">
        <RemolinoLogo />
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import RemolinoLogo from '~/assets/img/remolino.svg?inline';

@Component({
  components: {
    RemolinoLogo,
  },
})
export default class AboutMeInformation extends Vue {}

</script>

<style lang="scss" scoped>
.who-iam-section {
  display: flex;
  align-items: center;

  &__container {
    display: grid;
    grid-template-columns: 1fr 1fr;
    color: var(--text);
  }

  &__content {
    position: relative;
  }

  &__description {
    position: relative;
    z-index: 2;
    margin-top: rem(120);
    font-size: rem(18);
    line-height: 2;
  }

  &__title {
    position: absolute;
    top: 0;
    width: 100%;
    color: var(--asset-highlight);
    font-weight: bold;
    font-size: rem(182);
    line-height: 1;
    opacity: 0.5;
  }

  &__asset {
    display: flex;
    align-items: center;
    justify-content: center;

    & > svg {
      width: 90%;
      max-height: rem(300);
      fill: var(--asset-highlight);
    }
  }
}

.home {
  display: grid;
  grid-template-rows: 1fr;
  grid-template-columns: 1fr;
  align-items: center;
  justify-content: center;
  background-color: var(--background);

  &__content {
    padding: rem(92) rem(24);
    background-color: var(--background);

    &--secondary {
      background-color: var(--text-light);
    }

    &--tertiary {
      background-color: var(--background-footer);
    }
  }
}

@include mobile {
  .who-iam-section {
    &__container {
      grid-template-columns: 1fr;
    }

    &__asset {
      display: none;
    }
  }
}

</style>
