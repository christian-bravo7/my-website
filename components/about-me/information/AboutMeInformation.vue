<template>
  <section class="section-container">
    <div class="container | flex | my-0 mx-auto">
      <div class="text-black dark:text-white | about-me-information__content">
        <div class="relative | mb-4 md:mb-0">
          <h2 class="mb-4 | section-title about-me-information__title">
            {{ $t('me.hi-label') }}
          </h2>
          <p
            class="relative z-10 | font-base | about-me-information__description"
            v-html="$md($t('me.about-me-description', { age: currentAge }))"
          />
        </div>
        <div class="flex items-center justify-center">
          <img
            src="~assets/img/photo-2.png"
            alt="christian"
          >
        </div>
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

@Component
export default class AboutMeInformation extends Vue {
  readonly birthday: string = '2000-01-15';

  get currentAge (): number {
    const currentDate = new Date();
    const birth = new Date(Date.parse(this.birthday));

    const currentMonth = currentDate.getMonth();
    const currentYear = currentDate.getFullYear();
    const currentDay = currentDate.getDate();
    const birthMonth = birth.getMonth();
    const birthYear = birth.getFullYear();
    const birthDay = birth.getDate();

    const age = currentYear - birthYear;
    const ageNotCompleted = age - 1;

    if (birthMonth > currentMonth) {
      return ageNotCompleted;
    }

    if (birthMonth < currentMonth) {
      return age;
    }

    if (birthDay > currentDay) {
      return ageNotCompleted;
    }

    return age;
  }
}
</script>

<style lang="scss" scoped>
.about-me-information {
  &__content {
    @apply grid grid-cols-1 items-center;
  }

  &__title {
    @include theme-gradients(four, five);
  }

  &__description {
    /deep/ p {
      @apply mb-4;
    }
  }
}

@screen lg {
  .about-me-information {
    &__content {
      @apply gap-40;

      grid-template-columns: 1fr auto;
    }
  }
}
</style>
