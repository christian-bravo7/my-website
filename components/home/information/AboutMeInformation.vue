<template>
  <section class="section-container">
    <div class="container | flex justify-center | my-0 mx-auto">
      <div class="max-w-screen-lg | about-me-information__content">
        <div class="mb-4 md:mb-0">
          <h2 class="mb-8 | section-title about-me-information__title">
            {{ $t('me.about-me-label') }}
          </h2>
          <p
            class="font-base | about-me-information__description"
            v-html="$md($t('me.about-me-description'))"
          />
        </div>
        <div class="flex items-center justify-center">
          <img
            src="~assets/img/photo-2.png"
            alt="christian picture"
          >
        </div>
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

@Component
export default class AboutMeInformation extends Vue {}
</script>

<style lang="scss" scoped>
.about-me-information {
  &__content {
    @apply grid grid-cols-1 items-center;
  }

  &__title {
    @include theme-gradients(four, five);
  }

  &__description {
    /deep/ p {
      @apply mb-4;
    }
  }
}

@screen lg {
  .about-me-information {
    &__content {
      @apply gap-40;

      grid-template-columns: 2fr 1fr;
    }
  }
}
</style>
