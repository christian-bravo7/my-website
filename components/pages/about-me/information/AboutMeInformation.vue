<template>
  <section class="section-container">
    <div class="max-w-screen-lg | flex justify-center | my-0 mx-auto">
      <div class="about-me-information__content">
        <div class="mb-4 md:mb-0">
          <h2 class="mb-8 | section-title about-me-information__title">
            {{ $t('me.about-me-label') }}
          </h2>
          <p
            class="font-base | about-me-information__description"
            v-html="$md($t('me.about-me-description'))"
          />
        </div>
        <div class="flex items-center justify-center">
          <img
            class="w-full max-w-sm"
            src="~assets/img/photo-2.png"
            alt="christian picture"
          >
        </div>
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

@Component
export default class AboutMeInformation extends Vue {}
</script>

<style lang="scss" scoped>
.about-me-information {
  &__content {
    @apply grid grid-cols-1 lg:gap-12 items-center;
  }

  &__title {
    @include theme-gradients(four, five);
  }

  /* stylelint-disable */
  ::v-deep &__description {
    p {
      @apply mb-4;
    }
  }
}

@screen lg {
  .about-me-information {
    &__content {
      grid-template-columns: 2fr 1fr;
    }
  }
}
</style>
