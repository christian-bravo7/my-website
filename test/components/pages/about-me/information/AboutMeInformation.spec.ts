import { createLocalVue, mount } from '@vue/test-utils';

import AboutMeInformation from '@/components/pages/about-me/information/AboutMeInformation.vue';

import setupI18n from '@/test/i18n';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);

const componentConfig = {
  localVue,
  i18n,
};

describe('AboutMeInformation component', () => {
  it('should render correctly', () => {
    const wrapper = mount(AboutMeInformation, { ...componentConfig });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display title and description', () => {
    const wrapper = mount(AboutMeInformation, { ...componentConfig });

    const title = wrapper.find('h2');
    const descriptionText = wrapper.findAll('.about-me-information__description p');

    expect(descriptionText.at(0).text()).toBe('Hello, my name is Christian Bravo, a Frontend Developer from Guatemala with a great passion for web development using modern technologies.');
    expect(descriptionText.at(1).text()).toBe('I love solving problems, developing applications and creating dynamic, intuitive and accessible user interfaces.');
    expect(descriptionText.at(2).text()).toBe('Outside of the office you will find me playing soccer, collecting cactus, playing videogames and spending time with my pets and family.');

    expect(title.text()).toBe('About me');
  });
});
