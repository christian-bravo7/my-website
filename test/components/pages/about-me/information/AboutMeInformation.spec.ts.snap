// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMeInformation component should render correctly 1`] = `
<section class="section-container">
  <div class="max-w-screen-lg | flex justify-center | my-0 mx-auto">
    <div class="about-me-information__content">
      <div class="mb-4 md:mb-0">
        <h2 class="mb-8 | section-title about-me-information__title">
          About me
        </h2> <span><p class="font-base | about-me-information__description"><p>Hello, my name is Christian Bravo, a Frontend Developer from Guatemala with a great passion for web development using modern technologies.</p>
<p>I love solving problems, developing applications and creating dynamic, intuitive and accessible user interfaces.</p>
<p>Outside of the office you will find me playing soccer, collecting cactus, playing videogames and spending time with my pets and family.</p>
</p></span>
      </div>
      <div class="flex items-center justify-center"><img src="~assets/img/photo-3.png" alt="christian picture" class="w-full max-w-sm"></div>
    </div>
  </div>
</section>
`;
